src/signals/incident/components/form/MapSelectors/Caterpillar/CaterpillarLayer/CaterpillarLayer.tsx

```diff
@@ -1,42 +1,36 @@
 // SPDX-License-Identifier: MPL-2.0
 // Copyright (C) 2021 Gemeente Amsterdam
-import { useCallback, useContext, useMemo, useRef } from 'react'
+import { useCallback, useContext } from 'react'
 import L from 'leaflet'
 import { Marker } from '@amsterdam/arm-core'
 
 import type { FeatureCollection } from 'geojson'
 import type { FC } from 'react'
+import type { Item } from 'signals/incident/components/form/MapSelectors/Asset/types'
+import type { Feature } from 'signals/incident/components/form/MapSelectors/types'
+import type { Geometrie } from 'types/incident'
 
 import WfsDataContext from 'signals/incident/components/form/MapSelectors/Asset/Selector/WfsLayer/context'
 import SelectContext from 'signals/incident/components/form/MapSelectors/Asset/context'
-import type { Item } from 'signals/incident/components/form/MapSelectors/Asset/types'
-import type { Feature } from 'signals/incident/components/form/MapSelectors/types'
+import { featureTolocation } from 'shared/services/map-location'
 import { getIconUrl } from '../../utils'
 
 export const CaterpillarLayer: FC = () => {
   const { features } = useContext<FeatureCollection>(WfsDataContext)
-  const {
-    selection: selectionContext,
-    meta,
-    update,
-  } = useContext(SelectContext)
-  const selection = useRef<Item[]>(selectionContext)
-
-  selection.current = useMemo(() => selectionContext, [selectionContext])
+  const { selection, meta, setItem, removeItem } = useContext(SelectContext)
 
   const getMarker = useCallback(
     (feat: any) => {
       const feature = feat as Feature
-
-      const [lng, lat] = feature.geometry.coordinates
+      const coordinates = featureTolocation(feature.geometry as Geometrie)
       // Caterpillar layer renders only a single feature type (oak tree)
-      const featureType = meta.featureTypes[0]
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      const featureType = meta.featureTypes.find(
+        ({ label }) => label === 'Eikenboom'
+      )!
```

Comment:
Is this a safe way to identify the Eikenboom featureType? If anything from the feature type would change, it would be the label. I would suggest to take the typeValue

Reply:
Good catch. Very true! I'll change it.